app/src/main/cpp/skyline/services/fssrv/IFileSystem.cpp

```diff
@@ -1,8 +1,55 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
 
+#include <kernel/types/KProcess.h>
+#include <vfs/filesystem.h>
+#include "IFile.h"
 #include "IFileSystem.h"
 
 namespace skyline::service::fssrv {
-    IFileSystem::IFileSystem(FsType type, const DeviceState &state, ServiceManager &manager) : type(type), BaseService(state, manager, Service::fssrv_IFileSystem, "fssrv:IFileSystem", {}) {}
+    IFileSystem::IFileSystem(std::shared_ptr<vfs::FileSystem> backing, const DeviceState &state, ServiceManager &manager) : backing(backing), BaseService(state, manager, Service::fssrv_IFileSystem, "fssrv:IFileSystem", {
+        {0x0, SFUNC(IFileSystem::CreateFile)},
+        {0x7, SFUNC(IFileSystem::GetEntryType)},
+        {0x8, SFUNC(IFileSystem::OpenFile)},
+        {0xa, SFUNC(IFileSystem::Commit)}
+    }) {}
+
+    void IFileSystem::CreateFile(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        std::string path = std::string(state.process->GetPointer<char>(request.inputBuf.at(0).address));
```

Comment:
Yeah I think we should make a helper for this but it makes sense to do so together with nce3/refactor imo. Since there are other places that do the same